### PR TITLE
New weapon - I.C.E.E.

### PR DESCRIPTION
**Pull request by Stop-Signs**
## About the PR
Added a Cold Laser for epi to research. The cold laser, or I.C.E.E., deals 20 cold and reduces body temp. It will be a T3 in the arsenal tree.
## Why / Balance
More Content YAYYYYYYY. Adds some cool (pun intended) counterplay to the "speed is key" combat meta
## Technical details
TO DO
- [x] Make the projectile
- [x] Sprite the weapon (likely a recolor of the laser cannon
- [x]  tech tree
- [x] lathe recipe
- [x] Weapon code itself
- [x] Possibly more that im forgetting :3

## Media

Icon

## Requirements

- [x] I have tested all added content and changes.
- [x] I have added media to this PR or it does not require an ingame showcase.

## Credits
HUGE credit to OrksAreDaBest and deltanedas for helping me in C# 
**Changelog**

:cl:
- add: I.C.E.E, a laser that deals cold, is now available for epi to research

### CODE REVIEW

**Comment by Stop-Signs** [comment]
Right, so the main issue im dealing with is the lack of a way to make a hitscan laser modify body temp. if anyone knows how to do that please let me know.

**Comment by ThataKat** [comment]
I am totally talking off the top of my head here but have you looked into the ice wing watcher code? It's projectile isn't hitscan to my recollection, but it does exactly what you're talking about.

**Comment by Stop-Signs** [comment]
> I am totally talking off the top of my head here but have you looked into the ice wing watcher code? It's projectile isn't hitscan to my recollection, but it does exactly what you're talking about.

so that only works for projectiles, not hitscan. hitscan is shitcode and im currently working out a way to fix it

**Comment by Stop-Signs** [comment]
errors are from the sprites and i honestly dont know how to fix it

**Review by Radezolid — APPROVED** [review]
Looks good to me now.

File: Content.Server/Weapons/Ranged/Systems/GunSystem.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@
 using Content.Server.Interaction;
 using Content.Server.Power.EntitySystems;
 using Content.Server.Stunnable;
+using Content.Server.Temperature.Systems;
```
Comment (MilonPL) [inline Content.Server/Weapons/Ranged/Systems/GunSystem.cs]: DeltaV comment

File: Content.Server/Weapons/Ranged/Systems/GunSystem.cs [hunk 2]
```diff
@@ -40,6 +41,7 @@ public sealed partial class GunSystem : SharedGunSystem
     [Dependency] private readonly StaminaSystem _stamina = default!;
     [Dependency] private readonly StunSystem _stun = default!;
     [Dependency] private readonly SharedContainerSystem _container = default!;
+    [Dependency] private readonly TemperatureSystem _temperature = default!;
```
Comment (MilonPL) [inline Content.Server/Weapons/Ranged/Systems/GunSystem.cs]: DeltaV comment

File: Content.Shared/Weapons/Ranged/HitscanPrototype.cs [hunk 3]
```diff
@@ -16,6 +16,11 @@ public sealed partial class HitscanPrototype : IPrototype, IShootable
 
     [ViewVariables(VVAccess.ReadWrite), DataField("staminaDamage")]
     public float StaminaDamage;
+    /// <summary>
+    /// DeltaV: Changes the target's temperature by this amount when hit.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite), DataField("heatChange")]
```
Comment (MilonPL) [inline Content.Shared/Weapons/Ranged/HitscanPrototype.cs]: VV + tag in the datafield are redundant

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Projectiles/hitscan.yml [hunk 4]
```diff
@@ -13,3 +13,19 @@
...
+  muzzleFlash:
+    sprite: Objects/Weapons/Guns/Projectiles/projectiles.rsi
+    state: muzzle_omni
+  travelFlash:
+    sprite: Objects/Weapons/Guns/Projectiles/projectiles.rsi
+    state: beam_omni
+  impactFlash:
+    sprite: Objects/Weapons/Guns/Projectiles/projectiles.rsi
+    state: impact_omni
+  heatChange: -50000
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Projectiles/hitscan.yml]: EOF newline

**Review by MilonPL — CHANGES_REQUESTED** [review]

File: Content.Shared/Weapons/Ranged/HitscanPrototype.cs [hunk 5]
```diff
@@ -16,6 +16,11 @@ public sealed partial class HitscanPrototype : IPrototype, IShootable
 
     [ViewVariables(VVAccess.ReadWrite), DataField("staminaDamage")]
     public float StaminaDamage;
```
Comment (MilonPL) [inline Content.Shared/Weapons/Ranged/HitscanPrototype.cs:18]: missing a line break

File: Content.Server/Weapons/Ranged/Systems/GunSystem.cs [hunk 6]
```diff
@@ -4,6 +4,7 @@
 using Content.Server.Interaction;
 using Content.Server.Power.EntitySystems;
 using Content.Server.Stunnable;
+using Content.Server.Temperature.Systems;// DeltaV Heat Change system
```
Comment (MilonPL) [inline Content.Server/Weapons/Ranged/Systems/GunSystem.cs]: missing a space

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml [hunk 7]
```diff
@@ -261,3 +261,28 @@
     steps: 5
     zeroVisible: true
   - type: Appearance
+
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml:264]: only one linebreak here

File: Resources/Textures/DeltaV/Objects/Weapons/Guns/Battery/cold_cannon.rsi/meta.json [hunk 8]
```diff
@@ -0,0 +1,85 @@
+{
+	"version": 1,
+	"license": "CC-BY-SA-3.0",
+	"copyright": "Taken from vgstation at https://github.com/vgstation-coders/vgstation13/commit/125c975f1b3bf9826b37029e9ab5a5f89e975a7e, backpack sprite by Peptide, backpack sling sprite edited by Boaz1111, wield sprites by RiceMar1244",
```
Comment (MilonPL) [inline Resources/Textures/DeltaV/Objects/Weapons/Guns/Battery/cold_cannon.rsi/meta.json]: is this even the right commit lmao (needs to say it was resprited too)

**Review by MilonPL — CHANGES_REQUESTED** [review]

**Comment by CybersunOperative** [comment]
I like the concept of a cold laser. Gives security some added utility with a weapon that both slows and wounds targets. Seems fitting for a T3 damage wise. 

Other voices in direction have voiced that they like it as well. Direction approves :3

File: Content.Server/Weapons/Ranged/Systems/GunSystem.cs [hunk 9]
```diff
@@ -224,6 +226,12 @@ public override void Shoot(EntityUid gunUid, GunComponent gun, List<(EntityUid?
                     {
                         var hitEntity = lastHit.Value;
                         if (hitscan.StaminaDamage > 0f)
+
+                            _stamina.TakeStaminaDamage(hitEntity, hitscan.StaminaDamage, source: user);
```
Comment (MilonPL) [inline Content.Server/Weapons/Ranged/Systems/GunSystem.cs]: why

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml [hunk 10]
```diff
@@ -261,3 +261,27 @@
...
+- type: entity
+  name: I.C.E.E.
+  parent: [BaseWeaponBattery, BaseGunWieldable, BaseRestrictedContraband]
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml]: spaces with inline yaml lists `[ Entity1, Entity2 ]`

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml [hunk 11]
```diff
@@ -261,3 +261,27 @@
...
+  id: WeaponColdCannon
+  description: It's cooler than a normal gun.
+  components:
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml]: prototype order

**Review by MilonPL — CHANGES_REQUESTED** [review]

File: Content.Server/Weapons/Ranged/Systems/GunSystem.cs [hunk 12]
```diff
@@ -233,6 +235,11 @@ public override void Shoot(EntityUid gunUid, GunComponent gun, List<(EntityUid?
                     {
                         var hitEntity = lastHit.Value;
                         if (hitscan.StaminaDamage > 0f)
+
+                        // DeltaV: Changes the target's temperature by this amount when hit
+                        if (hitscan.HeatChange != 0f)
+                             _temperature.ChangeHeat(hitEntity, hitscan.HeatChange, true);
+
```
Comment (deltanedas) [inline Content.Server/Weapons/Ranged/Systems/GunSystem.cs]: this just broke the staminadamage thing, move this if statement below it

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml [hunk 13]
```diff
@@ -261,3 +261,27 @@
     steps: 5
     zeroVisible: true
   - type: Appearance
+
+- type: entity
+  parent: [ BaseWeaponBattery, BaseGunWieldable, BaseRestrictedContraband ]
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml]: should be security contraband incase it ever gets reenabled

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Review by Lyndomen — DISMISSED** [review]

**Review by deltanedas — APPROVED** [review]
ill fix nitpicks later

**Pull request merged**